src/main/java/seedu/opus/commons/util/FileUtil.java

```diff
@@ -4,19 +4,60 @@
 import java.io.IOException;
 import java.nio.file.Files;
 
+import seedu.opus.commons.exceptions.FileDeletionException;
+
 /**
  * Writes and reads files
  */
 public class FileUtil {
 
     private static final String CHARSET = "UTF-8";
 
-    public static boolean isFileExists(File file) {
+    //@@author A0148081H
+    public static void deleteFile(String filePath) throws FileDeletionException {
+        assert StringUtil.isValidPathToFile(filePath);
+
+        File file = new File(filePath);
+        if (!file.delete()) {
+            throw new FileDeletionException("Unable to delete file at: " + filePath);
+        }
+    }
+
+    /**
+     * Creates and deletes an empty file at the path.
+     * @param path must be a valid file path
+     * @return true if the path is exists and user has sufficient privileges.
+     */
+    public static boolean isPathAvailable(String path) {
+
+        File file = new File(path);
+        boolean exists = file.exists();
+
+        try {
+            createParentDirsOfFile(file);
+            file.createNewFile();
+        } catch (IOException e) {
+            return false;
+        }
+
+        if (!exists) { // prevent deleting an existing file
+            file.delete();
+        }
+        return true;
+    }
+
+    public static boolean doesFileExist(String filePath) {
```

Comment:
`does` feels a bit redundant here. `fileExists` will be a good enough method name here.